google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -3343,6 +3441,43 @@ private void createSessions(final int sessionCount, boolean distributeOverChanne
     }
   }
 
+  class MultiplexedSessionConsumer implements SessionConsumer {
+
+    @Override
+    public void onSessionReady(SessionImpl sessionImpl) {
+      final SettableFuture settableFuture = SettableFuture.create();
+      final MultiplexedSession newSession = new MultiplexedSession(sessionImpl);
+      settableFuture.set(newSession);
+
+      synchronized (lock) {
+        if (multiplexedSessionFuture.get() != null && multiplexedSessionFuture.get().isDone()) {
```

Comment:
I think that the first check `multiplexedSessionFuture.get() != null` will always be true and is redundant.

Reply:
Yes, removed.